app/models/node.rb

```diff
@@ -715,6 +725,27 @@ def self.new_note(params)
     [saved, node, revision]
   end
 
+  def self.new_preview_note(params)
+    author = User.find(params[:uid])
+    lat = params[:location][:latitude]
+    node = Node.new(uid:     author.uid,
+                    title:   params[:title],
+                    latitude: lat.to_f,
+                    longitude: params[:location][:longitude].to_f,
+                    comment: 2,
+                    type:    'note')
+    precision = node.decimals(lat.to_s)
+    node.precision = precision
+    revision = node.new_revision(uid:   author.uid,
+                                title: params[:title],
+                                body:  params[:body])
+    node.status = 4 if author.first_time_poster
```

Comment:
Maybe this line is not necessary as we're never saving this, so status shouldn't matter. Let's try removing it!
```suggestion
```